### ISSUE TITLE
FsCheck.Xunit fails generating immutable collections

### ISSUE THREAD

**Issue by AlexeyRaga** [fscheck/FsCheck#574]
I have the following type:

```csharp
public class TestRec
{
    public String Name { get; set; }
    public ImmutableDictionary<string, string> Headers { get; set; }
}
```

and writing a test like:

```csharp
[Property]
public async Task Test(TestRec message) { ... }
```

fails with an exception:

```
System.Exception
The type System.Collections.Immutable.ImmutableDictionary`2[[System.String, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e],[System.String, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]] is not handled automatically by FsCheck. Consider using another type or writing and registering a generator for it.
```

Since immutable collections are part of BCL. it would be nice to have them supported out of box.

**Comment by kurtschelfthout** [comment]
> Since immutable collections are part of BCL. it would be nice to have them supported out of box.

Don't they need a separate NuGet package? I've always avoided taking dependencies on non-core libraries just to add generators.

That said, it would be nice to have Arbitrary instances for these, I agree. Perhaps as a separate NuGet package?

**Comment by kurtschelfthout** [comment]
Released in 2.16.2
